File: src/main/java/me/andre111/d20server/scripting/variable/PropertyVariableSelf.java
package me.andre111.d20server.scripting.variable;

import me.andre111.d20common.model.Entity;
import me.andre111.d20server.scripting.Context;
import me.andre111.d20server.scripting.ScriptException;

public class PropertyVariableSelf extends PropertyVariable {
	public PropertyVariableSelf(String fullName, String propertyName) {
		super(fullName, propertyName);
	}

	@Override
	protected Entity getEntity(Context context) throws ScriptException {
		return context.self();
	}
}

File: core/common/scripting/variable/property-variable-self.js
import { PropertyVariable } from "./propertyvariable.js";

export class PropertyVariableSelf extends PropertyVariable {
    constructor(fullName, propertyName) {
        super(fullName, propertyName);
    }

    getEntity(context) {
        if(!context.self) throw new Error('No "self" entity in this context');
        return context.self;
    }
}
